.github/workflows/build_and_test_metrics_server.yml

```diff
@@ -0,0 +1,49 @@
+name: Build and Test - Metrics Server
+
+concurrency:
+  group: ${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+  push:
+    branches:
+      - master
+
+jobs:
+  metrics-server:
+    name: Metrics Server
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2.3.4
+
+      - name: Check if Metrics Server was modified
+        uses: dorny/paths-filter@v3
+        id: changes
+        with:
+          filters: |
+            src:
+              - 'src/metrics_server/**'
```

Comment:
What about changes to third_party and root files, like package.json and the pinned versions?

Reply:
The relevant `package.json` files are inside the relevant folders (i.e. `src/metrics_server/package.json`. I will add `third_party/**` to the shadowbox workflow though, since it's used there.